week07/이유라/todo/src/components/TodoView/TodoView.jsx

```diff
@@ -0,0 +1,99 @@
+import PropTypes from "prop-types";
+import { useState } from "react";
+
+const TodoView = ({ todos, deleteTodo, updateTodo, checkTodo }) => {
+  const [isEdited, setIsEdited] = useState(null);
+  const [editedContent, setEditedContent] = useState("");
+
+  const editTodo = (todo) => {
+    setIsEdited(todo.id);
+    setEditedContent(todo.content);
+  };
+
+  const editSaving = async (id, isComplete) => {
+    await updateTodo(id, editedContent, isComplete);
+    setIsEdited(null);
+  };
+
+  return (
+    <div className="w-[500px] h-[380px] mt-2 p-2 rounded-2xl bg-blue-400">
+      <p className="py-5 text-lg font-semibold">✨ Today&apos;s todo ✨</p>
+      <div className="h-[270px] overflow-scroll hide-scrollbar">
+        {todos.map((todo) => (
+          <div
+            key={todo.id}
+            className="w-4/5 flex justify-between items-center p-2.5 my-5 mx-auto "
+          >
+            {isEdited === todo.id ? (
+              <input
+                value={editedContent}
+                onChange={(e) => setEditedContent(e.target.value)}
+                className="text-lg font-semibold"
+                onClick={() => checkTodo(todo.id)}
+                checked={todo.isComplete}
+              />
+            ) : (
+              <>
+                <input
+                  type="checkbox"
+                  className="w-5 h-5 -ml-10 ppearance-none rounded-md focus:outline-none  "
+                  checked={todo.isComplete}
+                  onChange={() => checkTodo(todo.id)}
+                />
+                <p className="text-lg font-semibold -ml-5">{todo.content}</p>
+              </>
+            )}
+            <div>
+              {isEdited === todo.id ? (
+                <>
+                  <button
+                    className="bg-white border-none hover:bg-blue-100 px-4 py-1 rounded mr-4"
+                    onClick={() => editSaving(todo.id, todo.isComplete)}
+                  >
+                    저장
+                  </button>
```

Comment:
스타일링이 중복되고, `isEdited === todo.id` 상태에 따라 변경되기 때문에 코드를 줄여봤어요. 어떠신가요?

```suggestion
                  <button
                    className="bg-white border-none hover:bg-blue-100 px-4 py-1 rounded mr-4"
                    onClick={isEdited === todo.id? editSaving(todo.id, todo.isComplete) : deleteTodo(todo.id)}
                  >
                    {isEdited === todo.id ? '저장' : '삭제' }
                  </button>
```

Reply:
우와 감사합니다! 리뷰 반영해서 코드 수정하겠습니다